src/client/Client.cc

```diff
@@ -6372,6 +6372,12 @@ int Client::mount(const std::string &mount_root, const UserPerm& perms,
     }
   }
 
+  if(mdsmap->test_flag(CEPH_MDSMAP_REFUSE_CLIENT_SESSION)) {
```

Comment:
are we doing something similar for the kernel client?

Reply:
Yeah, we need to do the same in kclient after this PR get merged.

@dparmar18 Please raise one tracker for kclient for this to make sure we won't forget it.

Reply:
@lxbsz @ajarr sure, will do it.

Reply:
https://tracker.ceph.com/issues/58055 added.